tests/avifgridapitest.c

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//------------------------------------------------------------------------------
+
+// Creates an image where the pixel values are defined but do not matter.
+// Returns false in case of memory failure.
+static avifBool createImage(int width, int height, int depth, avifPixelFormat yuvFormat, avifImage ** image)
+{
+    *image = avifImageCreate(width, height, depth, yuvFormat);
+    if (*image == NULL) {
+        printf("ERROR: avifImageCreate() failed\n");
+        return AVIF_FALSE;
+    }
+    avifImageAllocatePlanes(*image, AVIF_PLANES_YUV);
+    if (width * height == 0) {
+        return AVIF_TRUE;
+    }
+
+    avifPixelFormatInfo formatInfo;
+    avifGetPixelFormatInfo((*image)->yuvFormat, &formatInfo);
+    uint32_t uvWidth = ((*image)->width + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX;
+    uint32_t uvHeight = ((*image)->height + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
+    if (uvWidth < 1) {
+        uvWidth = 1;
+    }
+    if (uvHeight < 1) {
+        uvHeight = 1;
+    }
+
+    for (uint32_t plane = 0; plane < (formatInfo.monochrome ? 1 : AVIF_PLANE_COUNT_YUV); ++plane) {
+        const uint32_t widthByteCount = ((plane == AVIF_CHAN_Y) ? (*image)->width : uvWidth) * (((*image)->depth > 8) ? 2 : 1);
+        const uint32_t planeHeight = (plane == AVIF_CHAN_Y) ? (*image)->height : uvHeight;
+        uint8_t * row = (*image)->yuvPlanes[plane];
+        for (uint32_t y = 0; y < planeHeight; ++y) {
+            memset(row, (int)(y % 256), widthByteCount); // Fill with a repeating gradient.
+            row += (*image)->yuvRowBytes[plane];
+        }
+    }
+    return AVIF_TRUE;
+}
+
+// Generates then encodes a grid image. Returns false in case of failure.
+static avifBool encodeGrid(int columns, int rows, int cellWidth, int cellHeight, int depth, avifPixelFormat yuvFormat, avifRWData * output)
+{
+    avifBool success = AVIF_FALSE;
+    avifEncoder * encoder = NULL;
+    avifImage ** cellImages = avifAlloc(sizeof(avifImage *) * columns * rows);
+    memset(cellImages, 0, sizeof(avifImage *) * columns * rows);
+    for (int row = 0, iCell = 0; row < rows; ++row) {
+        for (int col = 0; col < columns; ++col, ++iCell) {
```

Comment:
Nit: We can collapse the two for loops into one:

```
    for (int iCell = 0; iCell < columns * rows; ++iCell) {
```

Note that you do it this way under the `cleanup` label, at line 87.